src/NServiceBus.Core.Tests/Persistence/InMemory/SagaMetadataHelper.cs

```diff
@@ -1,26 +1,28 @@
 namespace NServiceBus.SagaPersisters.InMemory.Tests
 {
-    using System.Collections.Generic;
     using NServiceBus.Sagas;
 
     class SagaMetadataHelper
     {
-        public static IDictionary<string, object> GetMetadata<T>(IContainSagaData entity)
+        public static SagaCorrelationProperty GetMetadata<T>(IContainSagaData entity)
         {
             var metadata = SagaMetadata.Create(typeof(T));
 
-            var result = new Dictionary<string, object>();
-
-            foreach (var correlatedProp in metadata.CorrelationProperties)
+            SagaMetadata.CorrelationPropertyMetadata correlatedProp;
+            if (metadata.TryGetCorrelationProperty(out correlatedProp))
             {
                 var prop = entity.GetType().GetProperty(correlatedProp.Name);
 
                 var value = prop.GetValue(entity);
-
-                result[correlatedProp.Name] = value;
+                
+                return new SagaCorrelationProperty
+                {
+                    Name = correlatedProp.Name,
+                    Value = value
+                };
             }
 
-            return result;
+            return null;
```

Comment:
Since this is just there for the tests I chose the option that would mean the least amount of code?